maker/src/trading/mod.rs

```diff
@@ -0,0 +1,82 @@
+use crate::trading::orderbook_client::Direction;
+use crate::trading::orderbook_client::NewOrder;
+use crate::trading::orderbook_client::OrderResponse;
+use anyhow::Result;
+use bitcoin::Network;
+use futures::TryStreamExt;
+use reqwest::Url;
+use rust_decimal::Decimal;
+use rust_decimal_macros::dec;
+
+mod bitmex_client;
+mod orderbook_client;
+
+pub async fn run(orderbook_url: Url, maker_id: String, network: Network) -> Result<()> {
+    let network = match network {
+        Network::Bitcoin => bitmex_stream::Network::Mainnet,
+        _ => bitmex_stream::Network::Testnet,
+    };
+    let mut price_stream = bitmex_client::bitmex(network).await;
+
+    let mut last_bid = None;
+    let mut last_ask = None;
+
+    while let Some(quote) = price_stream.try_next().await? {
+        tracing::debug!("Received new quote {quote:?}");
+
+        last_bid = update_order(
```

Comment:
❓ I guess we are not adding a spread yet?

Reply:
Not yet :)